helper/transport.go

```diff
@@ -89,7 +92,12 @@ func (t *TransportWrapper) SetTLSConfig(c *tls.Config) error {
 }
 
 func (t *TransportWrapper) RoundTrip(req *http.Request) (*http.Response, error) {
+	transportID := uuid.New().String()
+	log.Printf("Request for UUID %s sent", transportID)
```

Comment:
Instead of logging the TXN ID here, I propose we use the `logReqMsg` and `logRespMsg` on lines 119 and 138, for example:

```golang
log.Printf("[DEBUG] "+logReqMsg, transportID, t.name, prettyPrintJsonLines(reqData))
```

and change the format strings:
```diff
-const logReqMsg = `%s API Request Details:
+const logReqMsg = `[%s] %s API Request Details:
```

```diff
-const logRespMsg = `%s API Response Details:
+const logRespMsg = `[%s] %s API Response Details:
```

Which would result in:
```
2024-02-14T14:37:32.659-0600 [DEBUG] provider.terraform-provider-vault: [2d01cac1-6c7c-48ca-b988-d20d9a9d2a29] Vault API Request Details:
---[ REQUEST ]---------------------------------------
GET /v1/sys/auth HTTP/1.1
...

2024-02-14T14:37:32.660-0600 [DEBUG] provider.terraform-provider-vault: [2d01cac1-6c7c-48ca-b988-d20d9a9d2a29] Vault API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/1.1 200 OK
...
```

Reply:
Cheers for the feedback! Definitely agree including with the const messages reads more gracefully. I've made the suggested changes - I'll close this convo and attach some sample output further down for reference.